patches.go

```diff
@@ -198,20 +130,108 @@ func ApplyCombineFromVariablesPatch(p PatchInterface, from, to runtime.Object) e
 	return errors.Wrap(patchFieldValueToObject(p.GetToFieldPath(), out, to), "cannot patch to object")
 }
 
-// IsOptionalFieldPathNotFound returns true if the supplied error indicates a
-// field path was not found, and the supplied policy indicates a patch from that
-// field path was optional.
-func IsOptionalFieldPathNotFound(err error, p *v1beta1.PatchPolicy) bool {
-	switch {
-	case p == nil:
-		fallthrough
-	case p.FromFieldPath == nil:
-		fallthrough
-	case *p.FromFieldPath == v1beta1.FromFieldPathPolicyOptional:
-		return fieldpath.IsNotFound(err)
-	default:
+// ApplyEnvironmentPatch applies a patch to or from the environment. Patches to
+// the environment are always from the observed XR. Patches from the environment
+// are always to the desired XR.
```

Comment:
I found that the `only` argument to `Apply` and `ApplyToObjects` was never used, so I deleted `filterPatch`.

At the same time I wanted to make `Render<T>Patches` operate on a single patch (not loop over all patches) so that I could have more fine grained handling of errors in `fn.go`.

At that point `Render<T>Patch` and `ApplyToObjects` were just two layers of switch, routing to the correct `Apply<T>Patch` function with the correct arguments. I figured it would be easier to follow if I merged the two layers of switch into one, which resulted in these new `ApplyEnvironmentPatch` and `ApplyComposedPatch` functions.